File: DSA Practice/LeetCode/24. Swap Nodes in Pairs/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ArrayList<Integer> arr = new ArrayList<Integer>();

        while(head != null){
            arr.add(head.val);
            head = head.next;
        }

        for (int i = 1; i < arr.size(); i+=2){
            int temp = arr.get(i);
            arr.set(i, arr.get(i-1));
            arr.set(i-1, temp);
        }

        ListNode temp = new ListNode();
        ListNode ref = temp;
        for (int i : arr){
            temp.next = new ListNode(i);
            temp = temp.next;
        }

        temp = ref;
        return temp.next;
    }
}

File: DSA Practice/LeetCode/24. Swap Nodes in Pairs/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        # Easy memory hogging method
        arr = []

        while(head):
            arr.append(head.val)
            head = head.next

        for i in range(1, len(arr), 2):
            arr[i-1], arr[i] = arr[i], arr[i-1]
        #print(arr)
        temp = ListNode()
        head = temp
        for i in arr:
            temp.next = ListNode(i)
            temp = temp.next
        temp = head
        return temp.next